allennlp/models/model.py

```diff
@@ -153,8 +156,19 @@ def forward_on_instances(self,
 
             instance_separated_output: List[Dict[str, numpy.ndarray]] = [{} for _ in dataset.instances]
             for name, output in list(outputs.items()):
+                # NOTE(markn): This is a hack because 0-dim pytorch tensors are not iterable.
+                # This occurs with batch size 1, because we still want to include the loss in that case.
+                if name == "loss" and output.dim() == 0:
```

Comment:
do you want to only do this for "loss", or for any 0-tensor?

Reply:
good point thanks